v2/internal/reconcilers/error_classifier.go

```diff
@@ -39,33 +24,35 @@ func stringOrDefault(str string, def string) string {
 	return str
 }
 
-func ClassifyCloudError(err *genericarmclient.CloudError) CloudErrorDetails {
+func ClassifyCloudError(err *core.CloudError) (core.CloudErrorDetails, error) {
```

Comment:
In this function we're always returning `error` as `nil`, do we still need to return an `error`? I think linter won't pass this.

Reply:
We need to match the signature of the extension point, even though this version can never error. 

I _hope_ it doesn't trigger the linter!